src/qinterface/protected.cpp

```diff
@@ -81,23 +81,20 @@ void _expLog2x2(complex* matrix2x2, complex* outMatrix2x2, bool isExp)
         complex trace = matrix2x2[0] + matrix2x2[3];
         complex determinant = (matrix2x2[0] * matrix2x2[3]) - (matrix2x2[1] * matrix2x2[2]);
         complex quadraticRoot =
-            sqrt((matrix2x2[0] - matrix2x2[3]) * (matrix2x2[0] - matrix2x2[3]) - (real1)(4.0) * determinant);
+            sqrt(trace * trace - (real1)(4.0) * determinant);
```

Comment:
Oh, was this math wrong the whole time?  It was supposed to be (m[0]+m[3])^2?

Reply:
When I noticed the bug, because I turned normalization off, I checked the math. There were a couple of little bugs, (but I swear this part was from a Wikipedia article). It passes with normalization off or on, now.